File: mycuration/src/main/java/com/phicdy/mycuration/presentation/view/CurationListView.java
package com.phicdy.mycuration.presentation.view;

import com.phicdy.mycuration.data.rss.Curation;

import java.util.ArrayList;

public interface CurationListView {
    void startEditCurationActivity(int editCurationId);
    void setNoRssTextToEmptyView();
    void setEmptyViewToList();
    void registerContextMenu();
    void initListBy(ArrayList<Curation> curations);
    void delete(Curation curation);
    int size();
    Curation curationAt(int position);
}

File: mycuration/src/main/java/com/phicdy/mycuration/presentation/view/CurationListView.kt
package com.phicdy.mycuration.presentation.view

import com.phicdy.mycuration.data.rss.Curation

import java.util.ArrayList

interface CurationListView {
    fun startEditCurationActivity(editCurationId: Int)
    fun setNoRssTextToEmptyView()
    fun registerContextMenu()
    fun initListBy(curations: ArrayList<Curation>)
    fun showRecyclerView()
    fun hideRecyclerView()
    fun showEmptyView()
    fun hideEmptyView()
}
